xdsl/dialects/builtin.py

```diff
@@ -1222,19 +1247,51 @@ def __init__(
 
         super().__init__([strides, offset])
 
-    def get_strides(self) -> Iterable[int | None]:
-        for stride in self.strides:
-            if isinstance(stride, NoneAttr):
-                yield None
-            else:
-                yield stride.data
+    def get_strides(self) -> Sequence[int | None]:
+        return [
+            None if isinstance(stride, NoneAttr) else stride.data
+            for stride in self.strides
+        ]
 
     def get_offset(self) -> int | None:
         if isinstance(self.offset, NoneAttr):
             return None
         else:
             return self.offset.data
 
+    def get_affine_map(self) -> AffineMap:
+        """
+        Return the affine mapping from the iteration space of this
+        layout to the element offset in linear memory. The resulting
+        affine map thus has only one result.
+
+        For dynamic strides, this results in an affinemap with a number
+        of symbols, ordered in the following manner:
+            (1) Symbols for every dynamic stride of the layout
+            (2) Symbol for the dynamic offset of the layout
```

Comment:
Is this MLIR's behaviour as well?

Reply:
I'll check this

Reply:
Oh, it seems to be the other way around: https://github.com/llvm/llvm-project/blob/56ad7cc0126f9899fd884391cfa10b6359206c01/mlir/lib/IR/BuiltinAttributes.cpp#L1790

I'll change it so it is the same